jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java

```diff
@@ -191,4 +208,47 @@ private static FileEntriesLayer addDirectoryContentsToLayer(
             });
     return builder.build();
   }
+
+  private static List<FileEntriesLayer> getDependenciesLayers(Path jarPath) throws IOException {
+    List<FileEntriesLayer> layers = new ArrayList<>();
+    String classPath = null;
+
+    // Get dependencies from Class-Path in the jar's manifest and add a layer each for non-snapshot
+    // and snapshot dependencies. If Class-Path is not present in the jar's manifest then skip
+    // adding the dependencies layers.
+    try (JarFile jarFile = new JarFile(jarPath.toFile())) {
+      classPath = jarFile.getManifest().getMainAttributes().getValue(Attributes.Name.CLASS_PATH);
+    }
+    if (classPath != null) {
+      Predicate<String> isSnapshot = name -> name.contains("SNAPSHOT");
+      List<String> allDependencies = Splitter.onPattern("\\s+").splitToList(classPath.trim());
+      List<Path> nonSnapshotDependencies =
+          allDependencies
+              .stream()
+              .filter(isSnapshot.negate())
+              .map(Paths::get)
+              .collect(Collectors.toList());
+      List<Path> snapshotDependencies =
+          allDependencies.stream().filter(isSnapshot).map(Paths::get).collect(Collectors.toList());
+      if (!nonSnapshotDependencies.isEmpty()) {
+        FileEntriesLayer.Builder nonSnapshotDependenciesLayerBuilder =
+            FileEntriesLayer.builder().setName(DEPENDENCIES);
+        nonSnapshotDependencies.forEach(
+            path ->
+                nonSnapshotDependenciesLayerBuilder.addEntry(
+                    path, APP_ROOT.resolve(RelativeUnixPath.get("dependencies")).resolve(path)));
```

Comment:
I think missed this. `jib jar` will fail if running from a different directory (e.g., `jib ... jar ../my.jar`). Perhaps the first argument to `addEntry` should be `jarPath.resolve(path)`?

Reply:
Ah great catch. To make sure that I'm understanding this correctly, would it need to be jarRoot.resolve(path) where jarRoot is the directory where the jar is located?

Reply:
I actually tried this. The command works at `~/A` but fails at `~`.

```
~/A$ pwd
/home/chanseok/A
~/A$ ls C.jar guava-20.0.jar 
C.jar  guava-20.0.jar
~/A$ jib --target=docker://foo jar C.jar
[WARN] Base image 'gcr.io/distroless/java' does not use a specific image digest - build may not be reproducible
Using base image with digest: sha256:90596f62a8ec0701d00845b85dfccf7f53b5e0bb3c49e275d7f19ea6c3e4565d

Container entrypoint set to [java, -cp, /app/explodedJar:/app/dependencies/*, C]
Executing tasks:
[==============================] 100.0% complete

~/A$ cd ..
~$ jib --target=docker://foo jar A/C.jar
[WARN] Base image 'gcr.io/distroless/java' does not use a specific image digest - build may not be reproducible
Using base image with digest: sha256:90596f62a8ec0701d00845b85dfccf7f53b5e0bb3c49e275d7f19ea6c3e4565d
Executing tasks:
[==================            ] 60.0% complete
> scheduling building manifests

java.nio.file.NoSuchFileException: guava-20.0.jar
```

And yeah, perhaps the source path should be something like `jarPath.getParent().resolve(path)`?